daemon/network.go

```diff
@@ -332,7 +332,19 @@ func (daemon *Daemon) createNetwork(cfg *config.Config, create types.NetworkCrea
 	}
 
 	if err := network.ValidateIPAM(create.IPAM, create.EnableIPv6); err != nil {
-		return nil, errdefs.InvalidParameter(err)
+		if agent {
+			// Swarm creates networks with agent=true to ensure that they exist on each agent
+			// when needed. Non-swarm networks created before ValidateIPAM was introduced
+			// continue to work, because they're not seen again here. By ignoring errors for
+			// agent networks here, swarm networks will continue to work too. (New overlay
+			// networks are initially created as !agent, so they are still validated.)
```

Comment:
When a swarm-scoped network is created it is run through `createNetwork` with `!agent`, but the network is not actually created by the `createNetwork` call. And this is true for all swarm-scoped networks irrespective of driver, not just overlay. (#32981)